src/contract/deploy/impl/SourceImpl.ts

```diff
@@ -28,18 +28,31 @@ export interface SourceData {
 
 export class SourceImpl implements Source {
   private readonly logger = LoggerFactory.INST.create('Source');
-  constructor(private readonly arweave: Arweave) {}
+  private readonly wallet: Wallet;
+
+  constructor(private readonly warp: Warp) {
+    this.wallet = new Wallet(this.warp);
+  }
 
   async save(
     contractData: SourceData,
     env: WarpEnvironment,
-    signer: ArWallet | SigningFunction,
+    wallet: ArWallet | Signature,
     useBundler = false
   ): Promise<any> {
     this.logger.debug('Creating new contract source');
 
     const { src, wasmSrcCodeDir, wasmGlueCode } = contractData;
 
+    this.wallet.getSignature(wallet);
```

Comment:
what's the point of this line?

Reply:
setting wallet property. User can use `save` method without deploying contract and the wallet instance should be then set from the begining. As an alternative I can probably pass the `wallet` of type Wallet in the deploy method and check in save method if wallet is of type Wallet or ArWallet | Signature. But it requires more 'ifs' and generally more complicated approach.